brave/brave-ios
# Fix #5405: Add origin w/ bolded eTLD+1 to Transaction Confirmation

## Summary of Changes
- Display origin w/ bolded eTLD+1 on Transaction Confirmation View. ERC20 Approval will also show the favicon.
- For brave wallet origin, we display 'Brave Wallet' with brave logo (when showing favicon)

This pull request fixes #5405 

## Submitter Checklist:


## Test Plan:

Brave Wallet ERC20 approve transaction:
1. Open & unlock wallet
2. Switch to Ropsten network
3. If you have 0 DAI, swap some ETH to DAI
4. Open swap panel and try to swap DAI to ETH. Swap button should show 'Activate Token DAI' if you have never swapped DAI before. Tap it.
5. Verify transaction shows brave logo + 'Brave Wallet' origin

Brave Wallet transaction:
1. Create a send/swap transaction within Brave Wallet
2. Verify transaction confirmation displays 'Brave Wallet' as origin

Dapp ERC20 approve transaction:
1. Open [Uniswap](https://app.uniswap.org/)
2. Change to swap from DAI to ETH (assuming never swapped DAI on Uniswap before, otherwise try another token)
3. Tap 'Allow the Uniswap Protocol to use your DAI'
4. Tap wallet notification
5. Verify transaction confirmation displays Uniswap favicon and the origin is displayed with a bolded eTLD+1

Dapp transaction:
1. Open [Uniswap](https://app.uniswap.org/)
6. Select DAI as the to token
7. Enter value in from field for ETH
8. Tap 'Swap' and then 'Confirm Swap'
9. Tap Wallet notification banner to open panel & transaction confirmation
10. Verify origin is displayed with a bolded eTLD+1

## Screenshots:

![brave wallet - erc20 approve](https://user-images.githubusercontent.com/5314553/171280389-7b793a53-6dea-4285-8417-161e631c32dd.png) | ![brave wallet - swap](https://user-images.githubusercontent.com/5314553/171280398-7be270ca-b88b-40b6-9b6f-cde634a0ddb9.png) | ![unswap - erc20 approve](https://user-images.githubusercontent.com/5314553/171280591-8a8bdf0f-5a4d-4826-91a7-5f43f4b7a9f7.png) | ![uniswap - swap](https://user-images.githubusercontent.com/5314553/171280605-1ee56aa1-720c-4549-b2fd-5bdc12e23fa8.png)
--|--|--|--


## Reviewer Checklist:

  - `QA/(Yes|No)`
  - `bug` / `enhancement`
